components/reference-data/src/main/java/org/datacleaner/beans/transform/SynonymLookupTransformer.java

```diff
@@ -90,7 +90,15 @@ public SynonymLookupTransformer(InputColumn<String> column, SynonymCatalog synon
 
     @Override
     public OutputColumns getOutputColumns() {
-        return new OutputColumns(String.class, new String[] { column.getName() + " (synonyms replaced)" });
+        if (lookUpEveryToken) {
+            return new OutputColumns(
+                    new String[] { column.getName() + " (synonyms replaced)", column.getName() + " (synonyms found)",
+                            column.getName() + " (master terms found)" },
+                    new Class[] { String.class, List.class, List.class });
```

Comment:
For most users it will probably be difficult to work with Lists. So I suggest we introduce a configured enum like ```RemoveDictionaryMatchesTransformer.RemovedMatchesType```.